packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -216,13 +247,13 @@ export const UploadControls = (): JSX.Element => {
     [direction, selection]
   );
 
-  return items && items.length > 0 ? (
+  return (
     <>
+      {fileSelect}
       <Title />
-      <Exit onClick={() => handleUpdateState({ type: 'EXIT' })} />
+      <Exit onClick={() => handleUpdateState({ type: 'CLEAR' })} />
       <Primary
         onClick={() => {
-          if (!items) return;
           handleUpload();
         }}
       >
```

Comment:
not blocking: Should the primary button be disabled after we start the upload action? At the moment if I have selected files, I can keep clicking the primary button and the network request gets sent each time even though the items already finished uploading

Reply:
Oh it definitely should be, will update